### DISCUSSION TITLE
issue:  use useFieldArray API like （append，move, etc）other field component rerender

### DISCUSSION THREAD

**Discussion by silverWolf818** [react-hook-form/react-hook-form#8329] [General]
### Version Number

7.30.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-usefieldarray-forked-bdfo6b?file=/src/index.js

### Steps to reproduce

1. click useFieldArray Button like （append，move, etc)
2. other field component rerender

### Expected behaviour

1. click useFieldArray Button like （append，move, etc)
2. other field component not rerender performance matters

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
That's expected behavior. react hook form would need to flush down a re-render at the root (useFieldArray) of your form.

**Reply by silverWolf818** [reply]
but other all controller field component all rerender ，poor performance

**Reply by silverWolf818** [reply]
<img width="386" alt="image" src="https://user-images.githubusercontent.com/7360502/167390760-c262d392-b51c-4743-b29f-acdd7d0f5e3b.png">

many no change component render poor performance

**Comment by silverWolf818** [comment]
I understand there is no problem with component rendered in useFieldArray returning fileds , but other fields that are not in this scope should not be rendered

**Reply by bluebill1049** [reply]
there is an feature request which you can upvote: https://github.com/react-hook-form/react-hook-form/issues/7472

**Reply by silverWolf818** [reply]
ok
